.github/workflows/test_e2e.yml

```diff
@@ -143,102 +146,89 @@ jobs:
 
       - uses: hetznercloud/setup-hcloud@v1
 
-      - uses: hetznercloud/tps-action@main
+      - uses: opentofu/setup-opentofu@v1
         with:
-          token: ${{ secrets.HCLOUD_TOKEN  }}
+          tofu_version: v1.7.2 # renovate: datasource=github-releases depName=opentofu/opentofu
+          tofu_wrapper: false
 
       - uses: yokawasa/action-setup-kube-tools@v0.11.1
         with:
           setup-tools: |
             helm
             kubectl
             skaffold
-          helm: v3.15.1
-          kubectl: v1.29.0
-          skaffold: v2.12.0
+          helm: v3.15.2 # renovate: datasource=github-releases depName=helm/helm
+          kubectl: v1.29.6 # renovate: datasource=github-releases depName=kubernetes/kubernetes
+          skaffold: v2.12.0 # renovate: datasource=github-releases depName=GoogleContainerTools/skaffold
+
       - name: Install k3sup
         run: |
           curl -sLS https://get.k3sup.dev | sh
 
-      - name: Setup test environment
+      - uses: hetznercloud/tps-action@main
+        with:
+          token: ${{ secrets.HCLOUD_TOKEN  }}
+
+      - name: Setup environment
         env:
+          ROBOT_SSH_KEY: ${{ secrets.ROBOT_SSH_KEY }}
           ROBOT_USER: ${{ secrets.ROBOT_USER }}
           ROBOT_PASSWORD: ${{ secrets.ROBOT_PASSWORD }}
         run: |
-          source <(hack/dev-up.sh)
-          
-          # make exported env variables available to following jobs
-          echo "KUBECONFIG=$KUBECONFIG" >> "$GITHUB_ENV"
-          echo "SKAFFOLD_DEFAULT_REPO=$SKAFFOLD_DEFAULT_REPO" >> "$GITHUB_ENV"
-          echo "CONTROL_IP=$CONTROL_IP" >> "$GITHUB_ENV"
-
-      - name: Build and Deploy HCCM
-        run: |
-          skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}"
-          tag=$(skaffold build --tag="e2e-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}" --quiet --output="{{ (index .Builds 0).Tag }}")
-          skaffold deploy \
-            --profile=robot \
-            --images=hetznercloud/hcloud-cloud-controller-manager=$tag
+          mkdir ~/.ssh # required for ssh-copy-id
+          install --mode=600 <(echo -n "$ROBOT_SSH_KEY") ./ssh_key
+          eval "$(ssh-agent -s)"
+          ssh-add ./ssh_key
+          trap "rm -f ./ssh_key; kill $SSH_AGENT_PID" EXIT
```

Comment:
Might be worth having a small wrapper since we use this in multiple places, e.g. `with-ssh-agent.sh`, that wraps the setup and teardown, and any command can be passed as argument.
 
That we could call like this: 
```
with-ssh-agent.sh make -C dev up
```

Reply:
I'll push a script.

Reply:
The new script was pushed, please update your local branch ;)

Reply:
Thanks!